python/amici/ode_export.py

```diff
@@ -1846,19 +1882,72 @@ def _compute_equation(self, name: str) -> None:
+            # fill boluses for Heaviside functions, as empty state updates
+            # would cause problems when writing the function file later
+            event_eqs = []
+            for event in self._events:
+                if event._state_update is None:
+                    event_eqs.append(sp.zeros(self.num_states_solver(), 1))
+                else:
+                    event_eqs.append(event._state_update)
+
+            self._eqs[name] = event_eqs
+
+        elif name == 'ddeltaxdx':
             self._eqs[name] = [
-                -self.eq('sroot')[ie, :] / self.eq('drootdt_total')[ie]
+                smart_jacobian(self.eq('deltax')[ie], self.sym('x'))
                 for ie in range(self.num_events())
             ]
 
-        elif name == 'deltasx':
+        elif name == 'ddeltaxdt':
+            self._eqs[name] = [
+                smart_jacobian(self.eq('deltax')[ie], time_symbol)
+                for ie in range(self.num_events())
+            ]
+
+        elif name == 'ddeltaxdp':
+            self._eqs[name] = [
+                smart_jacobian(self.eq('deltax')[ie], self.sym('p'))
+                for ie in range(self.num_events())
+            ]
+
+        elif name == 'stau':
             self._eqs[name] = [
-                smart_multiply((self.eq('xdot_old') - self.eq('xdot')),
-                               self.eq('stau')[ie])
+                -self.eq('sroot')[ie, :] / self.eq('drootdt_total')[ie]
                 for ie in range(self.num_events())
             ]
 
+        elif name == 'deltasx':
+            event_eqs = []
+            for ie, event in enumerate(self._events):
+                if event._state_update is not None:
+                    # ====== chain rule for the state variables ===============
+                    # get xdot with expressions back-substituted
+                    tmp_eq = smart_multiply(
+                        (self.sym('xdot_old') - self.sym('xdot')),
+                        self.eq('stau')[ie])
+                    # construct an enhanced state sensitivity, which accounts
+                    # for the time point sensitivity as well
+                    tmp_dxdp = self.sym('sx') * sp.ones(1, self.num_par())
+                    tmp_dxdp += smart_multiply(self.sym('xdot'),
+                                               self.eq('stau')[ie])
+                    tmp_eq += smart_multiply(self.eq('ddeltaxdx')[ie],
+                                             tmp_dxdp)
+                    # ====== chain rule for the time point ====================
+                    tmp_eq += smart_multiply(self.eq('ddeltaxdt')[ie],
+                                             self.eq('stau')[ie])
+                    # ====== partial derivative for the parameters ============
+                    tmp_eq += self.eq('ddeltaxdp')[ie]
+                else:
+                    tmp_eq = smart_multiply(
+                        (self.eq('xdot_old') - self.eq('xdot')),
```

Comment:
```suggestion
                        (self.sym('xdot_old') - self.eq('xdot')),
```

Reply:
This will do nothing, as for "xdot_old" eq == sym